src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -78,7 +78,7 @@ private ImmutableArray<ISymbol> GetSymbolsForCurrentContext()
                 // is a type-only context, we'll show all symbols anyway.
                 return GetSymbolsForExpressionOrStatementContext();
             }
-            else if (_context.IsTypeContext || _context.IsNamespaceContext)
+            else if (_context.IsTypeContext || _context.IsNamespaceContext || _context.IsLocalFunctionDeclarationContext)
```

Comment:
@CyrusNajmabadi This is probably naive or just plain wrong.  Do you have any objections? If not, I will add some more tests.

Reply:
I gave this a try in 842a1f6 and removed the new property in eb1b5be. This results in one test failure, discussed below.

Reply:
I tracked down every reference of `IsTypeContext` (at least I tried) and added tests for each occurrence.